tasks/agent.py

```diff
@@ -159,9 +159,17 @@ def build(ctx, rebuild=False, race=False, build_include=None, build_exclude=None
         "GOOS": "",
         "GOARCH": "",
     })
-    cmd = "go generate {}/cmd/agent"
+
+    cmd = "go generate -x {}/cmd/agent"
     ctx.run(cmd.format(REPO_PATH), env=env)
 
+    if with_both_python:
+        cmd = "go generate -x --run=\"agent-py2py3\" {}/cmd/agent"
+        ctx.run(cmd.format(REPO_PATH), env=env)
+    else:
+        cmd = "go generate -x --run=\"agent-py3\" {}/cmd/agent"
+        ctx.run(cmd.format(REPO_PATH), env=env)
+
```

Comment:
This is suboptimal, since we generate the `datadog.yaml` file three times:
- twice during the first `go generate`, which runs all `go:generate` directives, including `agent-py3` and `agent-py2py3`,
- plus one when we generate specifically the configuration that was requested.

To prevent this, during the first `go generate` we could exclude the directives with `agent-(py2py3|py3)` in their name. However, with the `--run` option you can decide to only run the directives which match the regex, but there's no way to exclude only the directives which match. Since negative lookaheads are also not available in Go regexes, there's no way here to exclude something easily.
The other way to do this would be to explicitly list all other directives, so that only them are run, but that could lead to surprises in the future if someone adds a directive.